Appraisals

```diff
@@ -1,7 +1,13 @@
-appraise 'rails_6_devise_4' do
+appraise 'rails_7_devise_4' do
   # use gemspec constraints
 end
```

Comment:
I believe we should be explicit here and define the versions here. In this case, because the `.gemspec` file contains the following line:

```Ruby
s.add_development_dependency "activerecord", ">= 3.2.6", "< 7"
```

we end up with Rails 6 being installed and tested in the `rails_7_device_4.gemfile.lock` file:


```
PATH
  remote: ..
  specs:
    simple_token_authentication (1.18.0)
      actionmailer (>= 3.2.6, < 8)
      actionpack (>= 3.2.6, < 8)
      active_model_serializers
      activemodel-serializers-xml
      devise (>= 3.2, < 6)

GEM
  remote: https://rubygems.org/
  specs:
    actionmailer (6.1.6)
      actionpack (= 6.1.6)
      actionview (= 6.1.6)
      activejob (= 6.1.6)
      activesupport (= 6.1.6)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.1.6)
      actionview (= 6.1.6)
      activesupport (= 6.1.6)
      rack (~> 2.0, >= 2.0.9)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actionview (6.1.6)
      activesupport (= 6.1.6)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    active_model_serializers (0.10.13)
      actionpack (>= 4.1, < 7.1)
      activemodel (>= 4.1, < 7.1)
      case_transform (>= 0.2)
      jsonapi-renderer (>= 0.1.1.beta1, < 0.3)
    activejob (6.1.6)
      activesupport (= 6.1.6)
      globalid (>= 0.3.6)
    activemodel (6.1.6)
      activesupport (= 6.1.6)
    activemodel-serializers-xml (1.0.2)
      activemodel (> 5.x)
      activesupport (> 5.x)
      builder (~> 3.1)
    activerecord (6.1.6)
      activemodel (= 6.1.6)
      activesupport (= 6.1.6)
    activesupport (6.1.6)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
      zeitwerk (~> 2.3)
    appraisal (2.4.1)
      bundler
      rake
      thor (>= 0.14.0)
    bcrypt (3.1.18)
    bson (4.15.0)
    builder (3.2.4)
    case_transform (0.2)
      activesupport
    coderay (1.1.3)
    concurrent-ruby (1.1.10)
    crass (1.0.6)
    devise (4.8.1)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0)
      responders
      warden (~> 1.2.3)
    diff-lcs (1.5.0)
    erubi (1.10.0)
    globalid (1.0.0)
      activesupport (>= 5.0)
    i18n (1.10.0)
      concurrent-ruby (~> 1.0)
    inch (0.8.0)
      pry
      sparkr (>= 0.2.0)
      term-ansicolor
      yard (~> 0.9.12)
    jsonapi-renderer (0.2.2)
    loofah (2.18.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    method_source (1.0.0)
    mini_mime (1.1.2)
    minitest (5.15.0)
    mongo (2.17.1)
      bson (>= 4.8.2, < 5.0.0)
    mongoid (7.4.0)
      activemodel (>= 5.1, < 7.1, != 7.0.0)
      mongo (>= 2.10.5, < 3.0.0)
      ruby2_keywords (~> 0.0.5)
    nokogiri (1.13.6-x86_64-darwin)
      racc (~> 1.4)
    orm_adapter (0.5.0)
    pry (0.14.1)
      coderay (~> 1.1)
      method_source (~> 1.0)
    racc (1.6.0)
    rack (2.2.3)
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.4.2)
      loofah (~> 2.3)
    railties (6.1.6)
      actionpack (= 6.1.6)
      activesupport (= 6.1.6)
      method_source
      rake (>= 12.2)
      thor (~> 1.0)
    rake (13.0.6)
    responders (3.0.1)
      actionpack (>= 5.0)
      railties (>= 5.0)
    rspec (3.11.0)
      rspec-core (~> 3.11.0)
      rspec-expectations (~> 3.11.0)
      rspec-mocks (~> 3.11.0)
    rspec-core (3.11.0)
      rspec-support (~> 3.11.0)
    rspec-expectations (3.11.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.11.0)
    rspec-mocks (3.11.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.11.0)
    rspec-support (3.11.0)
    ruby2_keywords (0.0.5)
    sparkr (0.4.1)
    sync (0.5.0)
    term-ansicolor (1.7.1)
      tins (~> 1.0)
    thor (1.2.1)
    tins (1.31.1)
      sync
    tzinfo (2.0.4)
      concurrent-ruby (~> 1.0)
    warden (1.2.9)
      rack (>= 2.0.9)
    webrick (1.7.0)
    yard (0.9.27)
      webrick (~> 1.7.0)
    zeitwerk (2.5.4)

PLATFORMS
  x86_64-darwin-20

DEPENDENCIES
  activerecord (>= 3.2.6, < 7)
  appraisal (~> 2.0)
  inch (~> 0.4)
  mongoid (>= 3.1.0, < 8)
  rspec (~> 3.0)
  simple_token_authentication!

BUNDLED WITH
   2.3.9
```

Reply:
This shouldn't be necessary... the gemspec constraints are meant to allow the latest release to be intalled.  But the gemspec constraint should be updated as part of adding support for Rails 7. (Those constraints are arguably what indicates that a given Rails version is suppported!)

Reply:
I see the gemspec was updated (probably since you comment @aried3r), based on visual review, I think we should be all right  on this side.